### PR TITLE
Add explain plan to dp_engine and test

### PR DESCRIPTION
**Pull request by jspacek**
## Description

- This PR changes `DPEngine` to use the `ReportGenerator` which constructs the explain plan for issue #10.

## Affected Dependencies
Part of #10 

## How has this been tested?
- `dp_engine_test` adds two reports and checks that they are valid
- Tested with `conda` environment under `Python 3.8`

## Checklist
- [x] I have followed the [Contribution Guidelines](https://github.com/OpenMined/.github/blob/master/CONTRIBUTING.md) and [Code of Conduct](https://github.com/OpenMined/.github/blob/master/CODE_OF_CONDUCT.md)
- [x] I have commented my code following the [OpenMined Styleguide](https://github.com/OpenMined/.github/blob/master/STYLEGUIDE.md)
- [x] I have labeled this PR with the relevant [Type labels](https://github.com/OpenMined/.github/labels?q=Type%3A)
- [x] My changes are covered by tests

### CODE REVIEW

File: pipeline_dp/dp_engine.py [hunk 1]
```diff
@@ -38,7 +43,22 @@ def aggregate(self, col, params: AggregateParams,
       data_extractors: functions that extract needed pieces of information from
         elements of 'col'
     """
-
+    if params is None:
+      return None
+    self._report_generators.append(ReportGenerator(params))
+    self._add_report_stage(f"Clip values to {params.low, params.high}")
```
Comment (dvadym) [inline pipeline_dp/dp_engine.py:49]: No need to add these stages now, they will be added when the code is implemented.

File: pipeline_dp/dp_engine.py [hunk 2]
```diff
@@ -27,9 +28,13 @@ def __init__(self, budget_accountant: BudgetAccountant,
...
     self._budget_accountant = budget_accountant
     self._ops = ops
+    self._report_generators = []
+
+  def _add_report_stage(self, text):
+    self._report_generators[-1].add_stage(text)
 
   def aggregate(self, col, params: AggregateParams,
-                data_extractors: DataExtractors):
+                data_extractors: DataExtractors):  # pylint: disable=unused-argument
```
Comment (dvadym) [inline pipeline_dp/dp_engine.py:37]: thanks for adding pylint comments!

**Review by dvadym — CHANGES_REQUESTED** [review]
Thanks, it looks good. I've left some a comment

File: pipeline_dp/dp_engine.py (hunk 2 above)
Comment (jspacek) [reply pipeline_dp/dp_engine.py:37]: 🍰 No problem!

File: pipeline_dp/dp_engine.py (hunk 1 above)
Comment (jspacek) [reply pipeline_dp/dp_engine.py]: 👍 Removed them and corresponding text based checks from the test (will just check for size of array).

**Review by dvadym — APPROVED** [review]
Thanks!

**Pull request merged**